packages/next/src/server/lib/recursive-readdir-sync.ts

```diff
@@ -13,13 +13,12 @@ export function recursiveReadDirSync(
   /** Used to replace the initial path, only the relative path is left, it's faster than path.relative. */
   rootDir = dir
 ): string[] {
-  const result = fs.readdirSync(dir)
+  const result = fs.readdirSync(dir, { withFileTypes: true })
 
-  result.forEach((part: string) => {
-    const absolutePath = join(dir, part)
-    const pathStat = fs.statSync(absolutePath)
+  result.forEach((part) => {
+    const absolutePath = join(dir, part.path)
```

Comment:
```suggestion
    const absolutePath = join(dir, part.name)
```

Should this be name instead of path?

Reply:
Well, this is just embarrassing :)